File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);

    set<int> numbers;
    int M; // 수행해야 하는 연산의 수
    cin >> M;

    for (int i = 0; i < M; i++) {
        string command;
        cin >> command;
        if (command == "add") {
            int number;
            cin >> number;
            numbers.insert(number);
        }
        else if (command == "remove") {
            int number;
            cin >> number;
            numbers.erase(number);
        }
        else if (command == "check") {
            int number;
            cin >> number;
            if (numbers.find(number) != numbers.end()) {
                cout << "1\n";
            }
            else { cout << "0\n"; }
        }
        else if (command == "toggle") {
            int number;
            cin >> number;
            if (numbers.find(number) != numbers.end()) {
                numbers.erase(number);
            }
            else { numbers.insert(number); }
        }
        else if (command == "all") {
            numbers
                    = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
                        11, 12, 13, 14, 15, 16, 17, 18, 19, 20 };
        }
        else if (command == "empty") {
            numbers.clear();
        }
    }
}

File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

M = int(input())
s = set()

for _ in range(M):
    command = input().split()
    if command[0] == "add":
        num = int(command[1])
        s.add(num)
    elif command[0] == "remove":
        num = int(command[1])
        s.discard(num)
    elif command[0] == "check":
        num = int(command[1])
        if num in s:
            print("1")
        else:
            print("0")
    elif command[0] == "toggle":
        num = int(command[1])
        if num in s:
            s.remove(num)
        else:
            s.add(num)
    elif command[0] == "all":
        s = set(range(1, 21))
    elif command[0] == "empty":
        s = set()